dashboard/app/views/home/_all_courses.html.haml

```diff
@@ -30,19 +28,49 @@
   wide_blocks = [
     Script::TWENTY_HOUR_NAME,
     "unplugged",
-    Script::EXPRESS_NAME,
-    Script::PRE_READER_EXPRESS_NAME,
+  ]
+
+  # Family name for a set of courses. A block will be generated for the recommended one.
+  family_wide_blocks = [
+    Script::EXPRESS,
+    Script::PREEXPRESS,
+  ]
+
+  # Family name for a set of courses. A block will be generated for the recommended one.
+  family_tall_blocks = [
+    Script::COURSEA,
+    Script::COURSEB,
+    Script::COURSEC,
+    Script::COURSED,
+    Script::COURSEE,
+    Script::COURSEF,
   ]
 
 .all-course-blocks
   - tall_blocks.each do |script_name|
     %div{id: script_name}
-      = render partial: 'shared/course_tall_block', locals: { id: script_name }
+      = render partial: 'shared/course_tall_block', locals: { id: script_name, family_name: nil}
 
   - wide_blocks.each do |script_name|
     %div{id: script_name}
       = render partial: 'shared/course_wide_block', locals: { id: script_name, view: "small" }
 
+  -# Creates a block for the recommended course version in the given course family.
+  - family_tall_blocks.each do |family_name|
+    - script = Script.get_script_family_redirect_for_user(family_name, user: user, locale: locale)
```

Comment:
i think we actually want to use [`Script.latest_stable_version`](https://github.com/code-dot-org/code-dot-org/blob/2cc61a885ad82dd79871bfab1656306182878bc8/dashboard/app/models/script.rb#L576-L600) here and on line 68 because:
1. we don't necessarily care about the user here, just the locale (correct me if i'm wrong though)
2. it returns the actual Script object instead of an empty Script object with a `redirect_to` property (we use those when a user attempts to go to an old script version but probably shouldn't be there)

Reply:
refactoring this will also help a little with the big jump in query counts on this page (as shown in `courses_controller_test.rb`), but i think we will want to refactor `Script.latest_stable_version` to use the `@@script_family_cache` (the helper method existed before the cache existed, i believe)

Reply:
I actually don't have much direction on this task. That absolute requirement was that we want students in Italy to get the 2018 courses because those are the latest. So the general idea was that we should just re-use the course redirect logic we already have. There's an argument for having the students see the course version that they have already started but it's not clear what the better experience is for students. For now, we can keep it simpler and just show the latest "stable" version as you have suggested
* I have updated the code to use `Script.latest_stable_version`
* I have updated `Script.latest_stable_version` to use `get_family_from_cache`.